grails-app/services/org/pih/warehouse/inventory/CycleCountTransactionService.groovy

```diff
@@ -42,42 +43,34 @@ class CycleCountTransactionService {
 
     private Transaction createAdjustmentTransaction(
             CycleCount cycleCount, Date transactionDate, boolean itemQuantityOnHandIsUpToDate) {
-        // We only care about the cycle count items from the most recent count. We create the entries first to avoid
-        // needlessly creating the transaction itself if there are no discrepancies.
+
+        // We need to compare the quantity counted against the most up to date QoH in product availability.
+        // However, if the cycle count items already have an up to date QoH (which will be the case if
+        // refreshQuantityOnHand == true when submitting the count), we don't need to re-compute it.
+        if (!itemQuantityOnHandIsUpToDate) {
```

Comment:
Shouldn't we recompute whether this is true instead of relying on an argument? Or is that what's happening within the caller when this is invoked?

Reply:
yeah it's happening in the caller. This will be true if command.refreshQuantityOnHand == true when submitting a count. We trust that the caller already did the refresh (cycleCountService.submitCount calls `cycleCountProductAvailabilityService.refreshProductAvailability(cycleCount)`) so that we can avoid doing it again here. It's a performance saving thing.

There isn't really another way of determining if this is true since to do so would require fetching product availability again (which is what we're trying to avoid).